### PR TITLE
Trigger node process when providerID populated

### PR DESCRIPTION
**Pull request by songrx1997**
/assign @swetharepakula

### CODE REVIEW

File: pkg/neg/controller.go [hunk 1]
```diff
@@ -319,6 +319,11 @@ func NewController(
 				logger.Info("Node has changed, enqueueing", "node", currentNode.Name)
 				negController.enqueueNode(currentNode)
 			}
+			// Trigger a sync when node providerID is populated.
+			if oldNode.Spec.ProviderID != currentNode.Spec.ProviderID {
+				logger.Info("Node was populated with provider id, enqueueing", "node", "providerID", currentNode.Name, currentNode.Spec.ProviderID)
```
Comment (swetharepakula) [inline pkg/neg/controller.go:324]: nit: Node provider ID changed

The parameters should be in the format "key1", "value1", "key2","value2"

File: pkg/neg/controller_test.go [hunk 2]
```diff
@@ -458,6 +458,53 @@ func TestEnqueueNodeWithILBSubsetting(t *testing.T) {
...
+	)
+	if err != nil {
+		t.Fatalf("Failed to create test node, error - %v", err)
+	}
+	nodeKey, err := cache.DeletionHandlingMetaNamespaceKeyFunc(node)
+	if err != nil {
+		t.Fatalf("Failed to create node key for test node %v, error - %v", node, err)
+	}
+	// sleep for the Informer store to pick this up.
+	time.Sleep(5 * time.Second)
```
Comment (swetharepakula) [inline pkg/neg/controller_test.go:491]: does this need to be 5 seconds?

File: pkg/neg/controller.go (hunk 1 above)
Comment (songrx1997) [reply pkg/neg/controller.go]: Done

File: pkg/neg/controller_test.go (hunk 2 above)
Comment (songrx1997) [reply pkg/neg/controller_test.go]: Change this to 1s.

File: pkg/neg/controller.go [hunk 3]
```diff
@@ -319,6 +319,11 @@ func NewController(
 				logger.Info("Node has changed, enqueueing", "node", currentNode.Name)
 				negController.enqueueNode(currentNode)
 			}
+			// Trigger a sync when node provider ID changed.
+			if oldNode.Spec.ProviderID != currentNode.Spec.ProviderID {
+				logger.Info("Node provider ID changed, enqueueing", "node", currentNode.Name, "providerID", currentNode.Spec.ProviderID)
```
[the remark this answers is not in the document]
Comment (songrx1997) [reply pkg/neg/controller.go:324]: The providerID can only be changed once from "" to a real string. The old one should always be a empty string here. What do you think?

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**